### PR TITLE
[EAP6-69] Modify the quickstart to make use of the new API to achieve this capability.

### PR DESCRIPTION
**Pull request by darranl**

### CODE REVIEW

File: ejb-security-interceptors/pom.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
 
         <version.jboss.maven.plugin>7.4.Final</version.jboss.maven.plugin>
 
-        <version.jboss.as>7.2.1.Final-redhat-10</version.jboss.as>
+        <version.jboss.as>7.4.0.Final-redhat-SNAPSHOT</version.jboss.as>
```
Comment (rafabene) [inline ejb-security-interceptors/pom.xml]: Shouldn't we release a -build version on jboss-developer.github.io/temp-maven-repo/ ?

**Comment by sgilda** [comment]
@rafabene is right. This is an API change, however. This change will break this for JBoss EAP 6.2. You need JBoss EAP 6.3 for this one.

@darranl , I noticed a few things:
-  These changes only work against EAP 6.3, correct? 
-  Should the README be updated to talk about the new API and also to note that it only works against JBoss EAP 6.3?

I am having difficulty getting this to compile. 
I believe the new module should be added to the http://jboss-developer.github.io/temp-maven-repo/ Maven repository
I had to changed the following property:

```
    <!--   <version.jboss.as>7.4.0.Final-redhat-SNAPSHOT</version.jboss.as> -->
    <version.jboss.as>7.4.0.Final-redhat-2</version.jboss.as>
```

Even with these changes, I get the following errors when I try to build the quickstart:

```
    [ERROR] The build could not read 1 project -> [Help 1]
    [ERROR]   
    [ERROR]   The project org.jboss.quickstarts.eap:jboss-ejb-security-interceptors:6.2.0-redhat-SNAPSHOT (/home/sgilda/GitRepos/jboss-eap-quickstarts/ejb-security-interceptors/pom.xml) has 2 errors
    [ERROR]     'dependencies.dependency.version' for org.jboss.as:jboss-as-security-api:jar is missing. @ line 90, column 21
    [ERROR]     'dependencies.dependency.version' for org.wildfly:wildfly-core-security-api:jar is missing. @ line 94, column 21
```

I also noticed the following in the POM file, which does not sound correct:

```
    <dependency>
        <groupId>org.wildfly</groupId>
        <artifactId>wildfly-core-security-api</artifactId>
    </dependency>
```

**Comment by sgilda** [comment]
Now I am getting this error on the build:

[ERROR]   The project org.jboss.quickstarts.eap:jboss-ejb-security-interceptors:6.2.0-redhat-SNAPSHOT (/home/sgilda/GitRepos/jboss-eap-quickstarts/ejb-security-interceptors/pom.xml) has 9 errors
[ERROR]     Non-resolvable import POM: Could not find artifact org.jboss.as:jboss-as-ejb-client-bom:pom:7.4.0.build-1 in jboss-developer-repository (http://jboss-developer.github.io/temp-maven-repo/) @ line 78, column 25 -> [Help 2]
[ERROR]     'dependencies.dependency.version' for org.jboss.as:jboss-as-security-api:jar is missing. @ line 90, column 21
[ERROR]     'dependencies.dependency.version' for org.picketbox:picketbox:jar is missing. @ line 104, column 21
[ERROR]     'dependencies.dependency.version' for org.wildfly:wildfly-core-security-api:jar is missing. @ line 109, column 21
[ERROR]     'dependencies.dependency.version' for org.jboss.ejb3:jboss-ejb3-ext-api:jar is missing. @ line 146, column 21
[ERROR]     'dependencies.dependency.version' for org.jboss:jboss-ejb-client:jar is missing. @ line 153, column 21
[ERROR]     'dependencies.dependency.version' for org.jboss.xnio:xnio-api:jar is missing. @ line 159, column 21
[ERROR]     'dependencies.dependency.version' for org.jboss.xnio:xnio-nio:jar is missing. @ line 165, column 21
[ERROR]     'dependencies.dependency.version' for org.jboss.marshalling:jboss-marshalling-river:jar is missing. @ line 172, column 21
[ERROR]

File: ejb-security-interceptors/pom.xml [hunk 2]
```diff
@@ -88,7 +88,32 @@
...
+                </exclusion>
+            </exclusions>
+        </dependency> 
+        <dependency>
+            <groupId>org.picketbox</groupId>
+            <artifactId>picketbox</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.wildfly</groupId>
```
Comment (sgilda) [inline ejb-security-interceptors/pom.xml:111]: Do we need wildfly dependencies with EAP 6.3?

Comment (darranl) [inline ejb-security-interceptors/pom.xml:111]: Yes this is the name of the module containing the new API

**Comment by sgilda** [comment]
@darranl : Got further this time, but got this error:

[ERROR] Failed to execute goal on project jboss-ejb-security-interceptors: Could not resolve dependencies for project org.jboss.quickstarts.eap:jboss-ejb-security-interceptors:jar:6.2.0-redhat-SNAPSHOT: Failed to collect dependencies for [org.jboss.as:jboss-as-security-api:jar:7.4.0.build-1 (compile), org.picketbox:picketbox:jar:4.0.17.SP2-redhat-2 (compile), org.wildfly:wildfly-core-security-api:jar:7.4.0.build-1 (compile), javax.enterprise:cdi-api:jar:1.0-SP4-redhat-2 (compile), org.jboss.spec.javax.annotation:jboss-annotations-api_1.1_spec:jar:1.0.1.Final-redhat-2 (compile), org.jboss.spec.javax.servlet:jboss-servlet-api_3.0_spec:jar:1.0.2.Final-redhat-1 (compile), org.jboss.spec.javax.ejb:jboss-ejb-api_3.1_spec:jar:1.0.2.Final-redhat-2 (compile), org.jboss.ejb3:jboss-ejb3-ext-api:jar:2.0.0-redhat-2 (compile), org.jboss:jboss-ejb-client:jar:1.0.23.Final-redhat-1 (compile), org.jboss.xnio:xnio-api:jar:3.0.7.GA-redhat-1 (runtime), org.jboss.xnio:xnio-nio:jar:3.0.7.GA-redhat-1 (runtime), org.jboss.marshalling:jboss-marshalling-river:jar:1.3.18.GA-redhat-1 (runtime), org.jboss.spec.javax.transaction:jboss-transaction-api_1.1_spec:jar:1.0.1.Final-redhat-2 (runtime)]: Failed to read artifact descriptor for org.jboss.as:jboss-as-security-api:jar:7.4.0.build-1: Could not find artifact org.jboss.as:jboss-as-security-parent:pom:7.4.0.build-1 in jboss-developer-repository (http://jboss-developer.github.io/temp-maven-repo/) -> [Help 1]

**Comment by darranl** [comment]
I will have another look later.

**Comment by sgilda** [comment]
This was merged.

**Pull request closed without merging**